### PR TITLE
sql: END alias for COMMIT, START for alias BEGIN.

### LINKED ISSUE #3861: add support for END (alias for COMMIT) and START TRANSACTION.
http://www.postgresql.org/docs/9.5/static/sql-end.html

### PR DESCRIPTION
**Pull request by dt**
Fixes #3861 

Still an open discussion on #3861 if this is something we want, but figured I'd get some concrete code in case.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3922)

### CODE REVIEW

File: sql/parser/sql.y [hunk 1]
```diff
@@ -1406,6 +1406,10 @@ transaction_stmt:
   {
     $$ = &CommitTransaction{}
   }
+| END opt_transaction
```
Comment (vivekmenezes) [inline sql/parser/sql.y]: change the above comment to // BEGIN / END / COMMIT / ROLLBACK

**Comment by vivekmenezes** [comment]
LGTM

**Comment by dt** [comment]
Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---

*[sql/parser/sql.y, line 1409 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922#-K8PmIB8ZgQjYtcrrcuX-r1-1409:-K8PnZGrD3cg-h4uwYe0:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/d6b1960fb34d9e39f97577a6268b7fa82cd1aeac/sql/parser/sql.y#L1409)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922)*

**Comment by petermattis** [comment]
It's good to add a parser specific test case to `sql/parser/parse_test.go:TestParse`. 

This code looks fine, but my concern about #3861 still stands. Why add support for `END` without adding support for `START`? Where do we draw the line on compatibility work like this?

---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922#-:-K8tSz-A0op01MnYhmL3:2090204836)*

**Comment by petermattis** [comment]
LGTM

---

Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/parser/parse_test.go, line 532 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922#-K8uFHKh9HeO0MU2MwRr:-K8uFHKh9HeO0MU2MwRs:1104587435) ([raw file](https://github.com/cockroachdb/cockroach/blob/b6c71dbe8cedaa95911a2daaf8c2df39df0f3fa7/sql/parser/parse_test.go#L532)):*
Note sure if I see a need for testing multiple statements like above. I'd just have cases like:

```
  {`BEGIN`, `BEGIN TRANSACTION`},
  {`START`, `BEGIN TRANSACTION`},
  {`END`, `COMMIT TRANSACTION`},
```

Btw, maybe for another PR, but we should `s/BEGIN TRANSACTION/START TRANSACTION/g`. That is, the SQL we output should be the standard by default, not an extension. 

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922#-:-K8uFkjW4Y5-D1hd2-eZ:676834233)*

**Comment by dt** [comment]
Review status: 0 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[sql/parser/parse_test.go, line 532 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922#-K8uFHKh9HeO0MU2MwRr:-K8uNECCmv0Guh_gWUbc:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/b6c71dbe8cedaa95911a2daaf8c2df39df0f3fa7/sql/parser/parse_test.go#L532)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3922)*

**Pull request merged**